source/ButtonModel.h

```diff
@@ -3,174 +3,100 @@
-	 */
-	virtual bool init() { return init(-1.0f); }
-
-	/**
-	 * Initializes a new button with the given angle
-	 *
-	 * An initializer does the real work that the constructor does not.  It
-	 * initializes all assets and makes the object read for use.  By separating
-	 * them, we allow ourselfs non-pointer references to complex objects.
+	 * Initializes a new button with the given angle and pair.
 	 *
 	 * @param a   The angle at which the button exists
+	 * @param pair Pointer to the pair of this button
+	 * @param pairID ID of the pair of this button
 	 *
 	 * @return true if the obstacle is initialized properly, false otherwise.
 	 */
-	virtual bool init(const float a) {
-		this->angle = a;
-		return true;
-	};
-
-	static std::shared_ptr<ButtonModel> alloc() {
-		std::shared_ptr<ButtonModel> result = std::make_shared<ButtonModel>();
-		return (result->init() ? result : nullptr);
-	}
+	bool init(const float a, std::shared_ptr<ButtonModel> pair, int pairID);
 
 #pragma mark -
 #pragma mark Accessors
 	/**
 	 * Returns the current angle of the button in degrees.
-	 *
-	 * @return the current angle of the button in degrees.
 	 */
 	float getAngle() { return angle; }
 
 	/**
-	 * Returns the current section of the button.
-	 *
-	 * @return the current section of the button.
+	 * Returns the section of the ship containing this button.
 	 */
 	int getSection();
 
 	/**
 	 * Returns the current height of the button.
-	 *
-	 * @return the current height of the button.
 	 */
 	int getHeight() { return height; }
 
 	/**
-	 * Returns the number of players in range of the button.
-	 *
-	 * @return the number of players in range of the button.
+	 * Adds the given player's flag to the button.
 	 */
-	int getPlayersOn() {
-		std::bitset<PLAYERS> ids(playersOn);
-		return (int)ids.count();
-	}
+	void addPlayer(int id) { playersOn.set(id); }
 
 	/**
-	 * Sets the current angle of the button in degrees.
-	 *
-	 * @param value The button angle in degrees
+	 * Removes the given player's flag from the button if present. Has no effect otherwise.
 	 */
-	void setAngle(float value) { angle = value; }
+	void removePlayer(int id) { playersOn.reset(id); }
 
 	/**
-	 * Adds the given player's flag from the button.
-	 *
+	 * Returns whether this button is resolved.
 	 */
-	void addPlayer(int id) { playersOn = playersOn | (unsigned char)pow(2, id); }
+	bool isResolved() { return resolved; }
 
 	/**
-	 * Removes the given player's flag from the button. Requires that this player is on the button
+	 * Resolve this button
 	 */
-	void removePlayer(int id) {
-		if (!isResolved()) {
-			playersOn = playersOn ^ (unsigned char)pow(2, id);
-		}
-	}
+	void resolve() { resolved = true; }
 
 	/**
-	 * Returns whether this player is on the button.
+	 * Returns whether any players are jumping on this button.
 	 */
-	bool isPlayerOn(int id) { return (playersOn & (unsigned char)pow(2, id)) > 0; }
+	bool jumpedOn() { return playersOn.any(); }
```

Comment:
This naminig is inconsistent with getter format `get__()` and the bool format `is__()`